x-pack/plugins/maps/public/connected_components/map_settings_panel/navigation_panel.tsx

```diff
@@ -4,25 +4,161 @@
-import { MAX_ZOOM, MIN_ZOOM } from '../../../common/constants';
+import { INITIAL_LOCATION, MAX_ZOOM, MIN_ZOOM } from '../../../common/constants';
+import { MapCenter } from '../../../common/descriptor_types';
+// @ts-ignore
+import { ValidatedRange } from '../../components/validated_range';
 
 interface Props {
+  center: MapCenter;
   settings: MapSettings;
   updateMapSetting: (settingKey: string, settingValue: string | number | boolean) => void;
+  zoom: number;
 }
 
-export function NavigationPanel({ settings, updateMapSetting }: Props) {
+const initialLocationOptions = [
+  {
+    id: INITIAL_LOCATION.LAST_SAVED_LOCATION,
+    label: i18n.translate('xpack.maps.mapSettingsPanel.lastSavedLocationLabel', {
+      defaultMessage: 'Map location at save',
+    }),
+  },
+  {
+    id: INITIAL_LOCATION.FIXED_LOCATION,
+    label: i18n.translate('xpack.maps.mapSettingsPanel.fixedLocationLabel', {
+      defaultMessage: 'Fixed location',
+    }),
+  },
+  {
+    id: INITIAL_LOCATION.BROWSER_LOCATION,
+    label: i18n.translate('xpack.maps.mapSettingsPanel.browserLocationLabel', {
+      defaultMessage: 'Browser location',
+    }),
+  },
+];
+
+export function NavigationPanel({ center, settings, updateMapSetting, zoom }: Props) {
   const onZoomChange = (value: Value) => {
     updateMapSetting('minZoom', Math.max(MIN_ZOOM, parseInt(value[0] as string, 10)));
     updateMapSetting('maxZoom', Math.min(MAX_ZOOM, parseInt(value[1] as string, 10)));
   };
 
+  const onInitialLocationChange = (optionId: string): void => {
+    updateMapSetting('initialLocation', optionId);
+  };
+
+  const onInitialLatChange = (event: ChangeEvent<HTMLInputElement>) => {
+    let value = parseFloat(event.target.value);
+    if (isNaN(value)) {
+      value = 0;
+    } else if (value < -90) {
+      value = -90;
+    } else if (value > 90) {
+      value = 90;
+    }
+    updateMapSetting('initialLat', value);
+  };
+
+  const onInitialLonChange = (event: ChangeEvent<HTMLInputElement>) => {
+    let value = parseFloat(event.target.value);
+    if (isNaN(value)) {
+      value = 0;
+    } else if (value < -180) {
+      value = -180;
+    } else if (value > 180) {
+      value = 180;
+    }
+    updateMapSetting('initialLon', value);
+  };
+
+  const onInitialZoomChange = (value: number) => {
+    updateMapSetting('initialZoom', value);
+  };
+
+  const useCurrentView = () => {
+    updateMapSetting('initialLat', center.lat);
+    updateMapSetting('initialLon', center.lon);
+    updateMapSetting('initialZoom', Math.round(zoom));
+  };
+
+  function renderInitialLocationInputs() {
+    if (settings.initialLocation === INITIAL_LOCATION.LAST_SAVED_LOCATION) {
+      return null;
+    }
+
+    const zoomFormRow = (
+      <EuiFormRow
+        label={i18n.translate('xpack.maps.mapSettingsPanel.initialZoomLabel', {
+          defaultMessage: 'Initial zoom',
+        })}
+        display="columnCompressed"
+      >
+        <ValidatedRange
```

Comment:
By introducing this, the min/max zoom-range settings now can break this input. Maybe validate the inputs there as well, so users cannot create an invalid config in the UX?

https://github.com/elastic/kibana/blob/f6b9b941aedc0907f8d4bc4ded42e0909a123d43/x-pack/plugins/maps/public/connected_components/map_settings_panel/navigation_panel.tsx#L57
e.g.: validate with something like
```
  const onZoomChange = (value: Value) => {
    let minZoom = parseInt(value[0] as string, 10);
    let maxZoom = parseInt(value[1] as string, 10);
    if (
      settings.initialLocation === INITIAL_LOCATION.FIXED_LOCATION ||
      settings.initialLocation === INITIAL_LOCATION.BROWSER_LOCATION
    ) {
      if (minZoom > settings.initialZoom) {
        minZoom = settings.initialZoom;
      }

      if (maxZoom < settings.initialZoom) {
        maxZoom = settings.initialZoom;
      }
    }
    updateMapSetting('minZoom', Math.max(MIN_ZOOM, minZoom));
    updateMapSetting('maxZoom', Math.min(MAX_ZOOM, maxZoom));
  };
```